xscen/extract.py

```diff
@@ -1223,6 +1240,36 @@ def _restrict_multimembers(catalogs: dict, id_columns: list, restrictions: dict)
     return catalogs
 
 
+def _restrict_wl(catalogs: dict, restrictions: dict):
+    """Update the results from search_data_catalogs by removing simulations that are not available in the warming level csv."""
+    df = pd.concat([catalogs[s].df for s in catalogs.keys()])
+    tas_csv = restrictions["tas_csv"]
+    if tas_csv is None:
+        tas_csv = Path(__file__).parent / "data/IPCC_annual_global_tas.csv"
+
+    # open csv
+    annual_tas = pd.read_csv(tas_csv, index_col="year")
+
+    if restrictions["ignore_member"]:
+        df["csv_name"] = df["mip_era"].str.cat(
+            [df["source"], df["experiment"]], sep="_"
+        )
+        csv_source = ["_".join(x.split("_")[:-1]) for x in annual_tas.columns[1:]]
+    else:
+        df["csv_name"] = df["mip_era"].str.cat(
+            [df["source"], df["experiment"], df["member"]], sep="_"
+        )
+        csv_source = list(annual_tas.columns[1:])
+
+    to_remove = pd.unique(df[~df["csv_name"].isin(csv_source)]["id"])
+
+    for k in to_remove:
+        logger.info(f"Removing {k} from the results.")
```

Comment:
If you follow my suggestion (`catalog` vs `catalogs`), I would remove this `logger` since it will potentially spam thousands of lines.

Also using `catalog`, I'd make `to_remove` a Series (or single row dataframe?) of booleans and subset based on that, which would be much faster than a `for` loop. Something like https://stackoverflow.com/questions/29281815/pandas-select-dataframe-columns-using-boolean

Reply:
Ok for subsetting  without the for loop. But, I think it would still be important to log the unique ids that are being removed.